crates/pep508-rs/src/lib.rs

```diff
@@ -976,12 +982,24 @@ fn parse(cursor: &mut Cursor, working_dir: Option<&Path>) -> Result<Requirement,
     // wsp*
     cursor.eat_whitespace();
     if let Some((pos, char)) = cursor.next() {
+        if let Some(VersionOrUrl::Url(url)) = requirement_kind {
+            // Unwrap safety: The `VerbatimUrl` we just parsed has a string source.
+            if url.given().unwrap().ends_with(';') && marker.is_none() {
+                return Err(Pep508Error {
+                    message: Pep508ErrorSource::String("Missing space before ';'".to_string()),
```

Comment:
This tells you what's gone wrong, but not why we need to issue an error here in the first place. I think if I were a user and I got this message, my immediate question would be, "Well, you obviously understood what I _meant_ here... so why not just do that?"

Maybe something like this? Not sure if there's a more concise way of explaining it:

```suggestion
                    message: Pep508ErrorSource::String("\
Missing space before ';' creates an ambiguous requirement
(Unclear whether the ';' is part of the URL or separates the URL from an environment marker)".to_string()
                    ),
```

Reply:
The reason itself is somewhat counterintuitive since we allow paths too, which unlike urls can contain spaces which we don't support except for the hack with file urls, and because PEP 508 doesn't require a semicolon after a version number for symmetry.